src/main/java/net/minecraftforge/common/crafting/CraftingHelper.java

```diff
@@ -582,20 +584,17 @@ public static void loadRecipes()
         //TODO: Figure out how to remove recipes, and override them. This relies on cpw to help.
         //For now this is only done one after mod init, I want to move this to ServerInit and re-do it many times.
         init();
+        ForgeRegistry<IRecipe> reg = (ForgeRegistry<IRecipe>)ForgeRegistries.RECIPES;
         if (DEBUG_LOAD_MINECRAFT)
-        {
-            Iterator<IRecipe> itr = GameData.getRecipeRegistry().iterator();
-            while(itr.hasNext())
-            {
-                itr.next();
-                itr.remove();
-            }
-        }
+            reg.clear();
+        else
+            GameData.revert(RegistryManager.FROZEN, GameData.RECIPES, false);
```

Comment:
There are already some modpacks (like FTB Beyond) which need at least 4GB of ram to even start (WTF?) and already taking at least 5 mins to load (loading thing multiple times, stiching multiple times, WTF?) because of these "oh, it's fine"/"is it a problem?"/etc.

So no, it's not fine. Yes, it is a problem!